web/src/sent_messages.js

```diff
@@ -15,44 +16,41 @@ export function get_new_local_id() {
     return "loc-" + local_id.toString();
 }
 
-function report_send_time(send_time, receive_time, locally_echoed, rendered_changed) {
-    const data = {
-        time: send_time.toString(),
-        received: receive_time.toString(),
-        locally_echoed,
-    };
-
-    if (locally_echoed) {
-        data.rendered_content_disparity = rendered_changed;
-    }
-
-    channel.post({
-        url: "/json/report/send_times",
-        data,
-    });
-}
-
 export class MessageState {
-    start = new Date();
+    local_id = undefined;
+    locally_echoed = undefined;
+    rendered_changed = false;
 
-    received = undefined;
-    send_finished = undefined;
-    rendered_content_disparity = false;
+    server_acked = false;
+    saw_event = false;
+
+    txn = undefined;
+    event_span = undefined;
 
     constructor(opts) {
         this.local_id = opts.local_id;
         this.locally_echoed = opts.locally_echoed;
     }
 
-    start_resend() {
-        this.start = new Date();
-        this.received = undefined;
-        this.send_finished = undefined;
-        this.rendered_content_disparity = false;
+    start_send() {
+        this.txn = Sentry.startTransaction({
+            op: "function",
+            description: "message send",
+            name: "message send",
+        });
+        this.event_span = this.txn.startChild({
+            op: "function",
+            description: "message send (server event loop)",
+        });
```

Comment:
The server response to our "send a message" HTTP request and the event loop acknowledgement can return in either order; how do the spans end up working there? Might make sense to write down in a comment somewhere in this file.

Reply:
OK I think what confused me here is that if we're going to track the span for the events return separately, it feels like we could also do that for getting the HTTP response back from the server, and this logic only precisely tracks the latter (and the end-to-end total as the transaction as a whole).

It looks like that was the preexisting behavior, so shouldn't block this PR, but would it be more natural to do a second `startChild` for the "message send (api response)"?